client/data/courses/use-course-query.js

```diff
@@ -5,7 +5,12 @@ const useCourseQuery = ( courseSlug, queryOptions = {} ) => {
 	return useQuery(
 		[ 'courses', courseSlug ],
 		() => wpcom.req.get( '/courses', { course_slug: courseSlug, apiNamespace: 'wpcom/v2' } ),
-		queryOptions
+		{
+			// Our course offering doesn't change that often, we don't need to
+			// re-fetch until the next page refresh.
+			staleTime: Infinity,
```

Comment:
👍 
Just checked out the video UI in My Home and saw that it refreshes every time the window is re-focused (which is default `react-query` behaviour). I think having the long cache time is better for this sort of data, it's not like user data we're trying to keep in-sync with the server.

The problem is that Calypso by default is persisting the data, which means using a `staleTime` of `Infinity` will literally never refresh the data. One option is to [add the `meta: { persist: false } }` option](https://github.com/Automattic/wp-calypso/blob/736beac709c61c2304721d845ca0df128d0dc795/client/data/README.md#L7). That means every page refresh will re-fetch the data. That's what I did for the design picker themes.

The other option is to set the `staleTime` to 24 hours or something.

I'm actually a bit conflicted over which is better, and I'm wondering whether I did the wrong thing for the design picker query.
- `staleTime = Infinity && persist = false` seems better for devs/a12s. Refreshing the data after F5 is better for the development flow.
- `staleTime = 24hrs` is better for users I think.

Reply:
> Just checked out the video UI in My Home and saw that it refreshes every time the window is re-focused (which is default react-query behaviour).

Yep, it's because it would re-fetching when the window is focused. If we want, we can disable `refetchOnWindowFocus` to avoid this behavior. But I don't see any re-fetching when the window is re-focused 🤔

> staleTime = Infinity && persist = false seems better for devs/a12s. Refreshing the data after F5 is better for the development flow.
> staleTime = 24hrs is better for users I think.

Hmmm...I think `persist = false` might be better as the user can get the latest data after refreshing the page. However, `staleTime = 24hr` still needs to wait the data to stale.

Reply:
I do see refetching when the UI gets focused, this seems unnecessary. 

Since we are doing a course management UI to edit videos descriptions and sort the video list I don't think that adding a `staleTime = 24hr` will do any good, thinking in the person crafting or updating the course. A better compromise is, in my opinion, `persist = false`.

Reply:
> I do see refetching when the UI gets focused, this seems unnecessary.

It's certainly interesting. It seems to be part of React Queries whole philosophy about fetch though and sort of actively trying to keep data from the server and client in sync. It probably works nicely for things where you're also auto-saving to the server. Not here though.

> A better compromise is, in my opinion, `persist = false`.

Agreed. Persisting can be nice to reduce the number of queries that get automatically fired off on page load. But given this request doesn't fire until an explicit user action (both on My Home and Hero Flow) it probably doesn't matter checking once per session.

Reply:
I think setting `refetchOnReconnect` to "always" might be more reliable to ensure re-fetch on page refresh. Sometimes, `persist=false` seems not to work if you don't set it along with `staleTime: Infinity` at the beginning.
I guess it's the reason why it doesn't work on @andres-blanco local environment: https://github.com/Automattic/wp-calypso/pull/59514#issuecomment-1000467500

Reply:
I've added a separate PR: https://github.com/Automattic/wp-calypso/pull/59576 to deal with the data caching. Let me know what you think :).

Reply:
LGTM 👍